### PR TITLE
Increase L0 SST size to 64MiB

### PR DESCRIPTION
**Pull request by palash25**
Fixes: #157

~~sorry, moving this to draft, i forgot to run the tests locally, i will mark this for review once i run them~~

`cargo test` passed locally

### CODE REVIEW

File: src/compactor_state.rs [hunk 1]
```diff
@@ -532,8 +533,20 @@ mod tests {
     }
 
     fn build_db(os: Arc<dyn ObjectStore>, tokio_handle: &Handle) -> Db {
+        let opts = DbOptions {
+            flush_interval: Duration::from_millis(100),
+            #[cfg(feature = "wal_disable")]
+            wal_enabled: true,
+            manifest_poll_interval: Duration::from_secs(1),
+            min_filter_keys: 1000,
+            l0_sst_size_bytes: 128,
```
Comment (palash25) [inline src/compactor_state.rs:542]: keeping this comment in mind https://github.com/slatedb/slatedb/pull/147#issuecomment-2308637068  which states ` our current default is something absurd like 128 bytes that's appropriate only for unit/integration tests` I've overriden the size to 128 by simply copying the Default config and changing the size back to 128 bytes here as the tests were failing with 64MB.

Let me know if I got it wrong and if I need to make these tests work with the default 64MB size.

I noticed that compactor.rs tests also use a config with 128 bytes but it had a few other fields different from the default configs so I wasn't sure if I should simply reuse that function by extracting it to some common file as I didn't want to break `compactor_state.rs` tests any further. Let me know if you would like to consider that as a possible solution to make both the unit tests pass with a default test config of sorts and reduce these two extra configs down to 1

Comment (criccomini) [reply src/compactor_state.rs:542]: Good call out. Let's leave `compactor.rs` as it is. I also like updating the test as you did to keep the 128 byte config. The only nit I have is that I _think_ you might be able to use this Rust syntax:

```rust
fn main() {
    let options = SomeOptions { foo: 42, ..Default::default() };
}
```

To inherit from default but override just the l0_sst_size_bytes.

**Comment by criccomini** [comment]
Looks good! Give the idiomatic Rust thing I mentioned a shot. If it doesn't work, no worries. Other than that, this is ready to merge. I'll wait for your final verdict on the `..Default`.

File: src/compactor_state.rs (hunk 1 above)
Comment (palash25) [reply src/compactor_state.rs:537]: oh thanks for that i had completely forgotten about this Rust syntax. Made the changes, it works

**Comment by criccomini** [comment]
Merged, thanks!

**Pull request merged**